source/Octopus.Cli/Commands/RunbookRun/DeleteRunbookRunsCommand.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Octopus.Cli.Infrastructure;
+using Octopus.Cli.Repositories;
+using Octopus.Cli.Util;
+using Octopus.Client;
+using Octopus.Client.Model;
+using Octopus.CommandLine;
+using Octopus.CommandLine.Commands;
+using Octopus.Versioning.Octopus;
+
+namespace Octopus.Cli.Commands.RunbooksRun
+{
+    [Command("delete-runbookruns", Description = "Deletes a range of runbook runs.")]
+    public class DeleteRunbookRunsCommand : RunbookRunCommandBase, ISupportFormattedOutput
+    {
+        List<RunbookRunResource> toDelete = new List<RunbookRunResource>();
+        List<RunbookRunResource> wouldDelete = new List<RunbookRunResource>();
+        DateTime minDate = new DateTime(1900, 1, 1);
+        DateTime maxDate = DateTime.MaxValue;
+
+        bool whatIf = false;
+
+        public DeleteRunbookRunsCommand(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFileSystem fileSystem, IOctopusClientFactory clientFactory, ICommandOutputProvider commandOutputProvider)
+            : base(repositoryFactory, fileSystem, clientFactory, commandOutputProvider)
+        {
+            var options = Options.For("Deletion");
+            options.Add<DateTime>("minCreateDate=", "[Optional] Earliest (inclusive) create date for the range of runbook runs to delete.", v => minDate = v);
+            options.Add<DateTime>("maxCreateDate=", "[Optional] Latest (inclusive) create date for the range of runbooks to delete.", v => maxDate = v);
+            options.Add<bool>("whatIf", "[Optional, Flag] if specified, releases won't actually be deleted, but will be listed as if simulating the command.", v => whatIf = true);
+        }
```

Comment:
Deletes are destructive and therefore we should have VERY restrictive inputs. I would prefer to see a mandatory project; dates may be far too inclusive for operations like this.

Reply:
I'll add the mandatory project argument as you requested above; are you also suggesting making the min and max date required?

Reply:
Made the minCreateDate and maxCreateDate arguments required as well.